### ISSUE TITLE
Enhancement: js error Uncaught TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by ekrichbaum** [NagVis/nagvis#240]
related to weathermap lines.  1.9.17 version.
In the renderLabels function there are two lines that call renderLabel

```
this.renderLabel(0);
this.renderLabel(1);

```
In some cases, the renderLabel gets an object with no .parts causing the error. I wasn't able to find what caused the values to be empty but fixed the error in renderLabel with a check for the array.  Also moved the dump out for perfdata being null up to the top of the function.  No need to stay if the data isn't there.

```
renderLabel: function(id) {
if (this.parts.length === 0) return;
if (this.perfdata === null) return;

```
In both NagVisCompressed.js and ElementLine.js.

Doesn't fix how it got there but drops out prior to tripping the visual error.  Could also drop a console.log(this); in before the return to see what objects but that didn't really lead me anywhere.

**Comment by ekrichbaum** [comment]
#241

**Comment by LarsMichelsen** [comment]
Has been fixed by merging #241
